lib/renderer/src/renderer/ResourceStorage.h

```diff
@@ -0,0 +1,86 @@
+//
+// Created by nikita on 26.11.2021.
+//
+
+#ifndef VANGERS_RIDSTORAGE_H
+#define VANGERS_RIDSTORAGE_H
+
+#include <unordered_map>
+#include <memory>
+
+#include "common.h"
+#include "exception.h"
+
+namespace renderer {
+	template<typename TResourceId, typename TResource>

```

Comment:
why is this `TResourceId` at all? Wouldn't it be the same as `Resource<TResource>`?

Reply:
Yes, it's just like that because I'm not totally happy with my current unique integral type approach. By *unique integral type* I mean that we have integer IDs, but we should not pass a Texture ID to the HeightMap methods for example. So I keep the ResourceStorage accepting any struct with `int32_t id` field in case if I change the current approach.

Reply:
We can cross this bridge when we get there. Seems strange to overcomplicate the API for something we don't know, at this point. Your choice though, not a big deal.